narwhal/primary/src/aggregators.rs

```diff
@@ -57,11 +64,36 @@ impl VotesAggregator {
 
         if self.weight >= committee.quorum_threshold() {
             self.weight = 0; // Ensures quorum is only reached once.
-            return Ok(Some(Certificate::new(
-                committee,
-                header.clone(),
-                self.votes.clone(),
-            )?));
+            let cert = Certificate::new(committee, header.clone(), self.votes.clone())?;
+            let (_, pks) = cert.signed_by(committee);
+
+            let certificate_digest: Digest<{ crypto::DIGEST_LENGTH }> = Digest::from(cert.digest());
+            match cert
+                .aggregated_signature
+                .verify_secure(&to_intent_message(certificate_digest), &pks[..])
+            {
+                Err(_) => {
```

Comment:
Can we log the error as well or is it not very informative?

Reply:
its most likely to be an opaque "invalid signature error" here, but since it can potentially have invalid length etc, we can log it